scripts/gen-dns-sync

```diff
@@ -34,18 +37,96 @@ while [[ -z "$HUBIP" ]]; do
 done
 [[ $hdr == 1 ]] && echo " done."
 
-# See if we already have the correct entry, skip if so
-CURIP=$(doctl compute domain records list $DOMAIN -o json | jq -r ".[] | select( .name == \"$HOST\") | .data")
-if [[ "$CURIP" == "$HUBIP" ]]; then
-	echo "DNS for $HUB_FQDN already points to the correct IP ($CURIP)."
-	exit
-fi
-
-# Delete any old DNS entries
-DNS_RECORD_IDS=$(doctl compute domain records list $DOMAIN -o json | jq ".[] | select( .name == \"$HOST\") | .id ")
-for ID in $DNS_RECORD_IDS; do
-	doctl compute domain records delete $DOMAIN $ID -f
-done
+if [[ $PROVIDER == do ]]; then
+	# See if we already have the correct entry, skip if so
+	CURIP=$(doctl compute domain records list $DOMAIN -o json | jq -r ".[] | select( .name == \"$HOST\") | .data")
+	if [[ "$CURIP" == "$HUBIP" ]]; then
+		echo "DNS for $HUB_FQDN already points to the correct IP ($CURIP)."
+		exit
+	fi
+
+	# Delete any old DNS entries
+	DNS_RECORD_IDS=$(doctl compute domain records list $DOMAIN -o json | jq ".[] | select( .name == \"$HOST\") | .id ")
+	for ID in $DNS_RECORD_IDS; do
+		doctl compute domain records delete $DOMAIN $ID -f
+	done
+
+	# Create a new DNS entry pointing to the external IP
+	doctl compute domain records create $DOMAIN --record-type=A --record-name=$HOST --record-data=$HUBIP --record-ttl 30
+elif [[ $PROVIDER == aws ]]; then
+	hosted_zone_id=$(aws --output=json route53 list-hosted-zones | jq --arg DOMAIN "${DOMAIN}." ' .HostedZones | .[] | select(.Name==$DOMAIN) | .Id ')
+	# remove trailing / preceeding quotes
+	hosted_zone_id="${hosted_zone_id%\"}"
+	hosted_zone_id="${hosted_zone_id#\"}"
+
+	resources=$(aws --output=json route53 list-resource-record-sets --hosted-zone-id $hosted_zone_id)
+	record=$(echo $resources | jq --arg  HUB_FQDN "${HUB_FQDN}." ' .ResourceRecordSets | .[] | select(.Name==$HUB_FQDN and .Type=="A") ')
```

Comment:
You should quote the `$resources` variable here (i.e., `echo "$resources"`).

Here's why:
```
$ foo="{ json:    \"some    thing\";    }"
$ echo "$foo"
{ json:    "some    thing";    }
$ echo $foo
{ json: "some thing"; }
```
Note how you lost the spaces in the `"some    thing"` string in the second invocation.

Check your files for similar calls (e.g. search for instances of `echo $`) and fix if necessary. bash is a cruel mistress.

Reply:
Made that change throughout